### PR TITLE
Use get_release_names to avoid timeouts on Rule view

### PR DESCRIPTION
**Pull request by gabrielBusta**
#2785 

cc @bhearsum

### CODE REVIEW

File: src/auslib/web/admin/swagger/api_v2.yml [hunk 1]
```diff
@@ -463,8 +478,9 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/ReleaseModifyResponse'
-                $ref: '#/components/schemas/ReleasePinNotSetResponse'
+                anyOf:
+                  - $ref: '#/components/schemas/ReleaseModifyResponse'
+                  - $ref: '#/components/schemas/ReleasePinNotSetResponse'
```
Comment (bhearsum) [inline src/auslib/web/admin/swagger/api_v2.yml:483]: This is an unrelated fix, I assume?

File: ui/src/views/Rules/Rule/index.jsx [hunk 2]
```diff
@@ -170,7 +170,9 @@ function Rule({ isNewRule, user, ...props }) {
     Promise.all([fetchReleaseNames(), fetchReleaseNamesV2()]).then(
       ([names, namesV2]) => {
         if (names.data.data && namesV2.data) {
-          setReleaseNames(names.data.data.names.concat(namesV2.data).sort());
+          setReleaseNames(
+            names.data.data.names.concat(namesV2.data.data).sort()
```
Comment (bhearsum) [inline ui/src/views/Rules/Rule/index.jsx:174]: 3 names and 4 datas in one line, wow! It might be worth doing some of the parsing in `services/releases.js` instead (as we do in the code you're removing) -- but that's not at all a blocker.

File: src/auslib/web/admin/swagger/api_v2.yml [hunk 3]
```diff
@@ -463,8 +478,9 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/ReleaseModifyResponse'
```
Comment (gabrielBusta) [inline src/auslib/web/admin/swagger/api_v2.yml:466]: Not sure about this one, but having a non-unique key in an object is not valid YAML

**Review by bhearsum — APPROVED** [review]
Thank you for getting this together so quickly! One thing you could optionally improve below, but it's definitely not a blocker.

File: src/auslib/web/admin/swagger/api_v2.yml [hunk 4]
```diff
@@ -463,8 +478,9 @@ paths:
           content:
             application/json:
               schema:
```
Comment (gabrielBusta) [inline src/auslib/web/admin/swagger/api_v2.yml:480]: 🤷❓afik having a non-unique key in an object is not valid YAML

File: src/auslib/web/admin/swagger/api_v2.yml (hunk 1 above)
Comment (gabrielBusta) [reply src/auslib/web/admin/swagger/api_v2.yml:483]: Yup, a nit-picky one at that. The Swagger parser is complaining that `Map keys must be unique` I try to load the API spec in the Swagger editor. I opted for `anyOf` because it satisfies the test cases for this end-point

File: src/auslib/web/admin/swagger/api_v2.yml [hunk 5]
```diff
@@ -463,15 +478,32 @@ paths:
...
+                anyOf:
+                  - $ref: '#/components/schemas/ReleaseModifyResponse'
+                  - $ref: '#/components/schemas/ReleasePinNotSetResponse'
         '400':
           description: Client error
         '403':
           description: Permission denied
 
   /pins/{product}/{channel}/{version}:
     get:
```
Comment (gabrielBusta) [inline src/auslib/web/admin/swagger/api_v2.yml:490]: When I loaded the API spec into the Swagger editor it found a few semantic errors on this end-point because "Declared path parameter "foo" needs to be defined as a path parameter at either the path or operation level"

**Comment by bhearsum** [comment]
Whoops, I didn't realize @jcristau had been requested to review this, heh. I still think he ought to look at it before it lands, as he's much more up to date on Balrog than me.

**Comment by jcristau** [comment]
Thanks for tackling this.

I sort of feel like this is making the API spec more complex, and having a single API endpoint to get both v1 and v2 release names, using the existing code in `auslib.web.common.releases.get_releases`, or at least somehow sharing code with the public `/api/v1/releases`, would be simpler all around: the server-side code already exists since it is used by the public endpoint, and the frontend would just need to do a single xhr instead of two.  Is there a particular reason not to go that way?
Also: AIUI we're setting the swagger version to 2.0, and there's a few preexisting comments about oneOf only being supported in 3.0, and so working around that by listing the different possible properties without any of them being "required".  Likewise anyOf doesn't seem to be in the 2.0 version.

One nit: use a separate commit for fixing existing API spec bugs

And one possible followup: can we somehow catch/expose the issues you found in the swagger spec in CI?

**Comment by gabrielBusta** [comment]
I ended up just sticking with the query param because it's more symmetrical with what is already there so it's a smaller refactor.

We are using openapi 3.0 on the v2 file, I am going to clean up v1 too.

I found a way to test the spec in CI https://github.com/APIDevTools/swagger-cli

CI check coming soon
